apps/glusterfs/health_cache.go

```diff
@@ -171,6 +172,10 @@ func (hc *NodeHealthCache) toProbe() ([]*NodeHealthStatus, error) {
 
 func (s *NodeHealthStatus) update(e executors.Executor) {
 	// TODO: add ability to skip check if node was already recently checked
+
+	if !s.Up {
+		return
+	}
```

Comment:
This doesn't make any sense to me. If the node is up and then goes down later how will ever detect that?

Reply:
updated code,  only if the node is marked as up, then only checking the glusterd status or else just  updating the state

Reply:
I'm still confused. In addition this appears to be a conditional that does nothing. Is the current version complete?